### PR TITLE
Nullary Relation and bin hex

### PR DESCRIPTION
**Pull request by psubotic**
Nulary relations and bin hex

### CODE REVIEW

File: src/AstRelation.h [hunk 1]
```diff
@@ -123,9 +123,12 @@ class AstRelation : public AstNode {
     /** Print string representation of the relation to a given output stream */
     virtual void print(std::ostream &os) const {
         os << ".decl " << this->getName() << "(";
-        os << attributes[0]->getAttributeName() << ":" << attributes[0]->getTypeName();
-        for (size_t i=1; i<attributes.size(); ++i) {
-            os << "," << attributes[i]->getAttributeName() << ":" << attributes[i]->getTypeName();
+        if (attributes.size() > 0) {
```
Comment (HerbertJordan) [inline src/AstRelation.h]: this part could be reduced to a single line by using the `join` utility, which lists elements with a seperator

File: src/RamRelation.h [hunk 2]
```diff
@@ -93,6 +100,10 @@ class RamRelationIdentifier {
         return input;
     }
 
+    bool isNullary() const { 
+        return nullary;
```
Comment (HerbertJordan) [inline src/RamRelation.h]: I thinki it would be nicer if this is taken from the number of attributes, not an extra field.

**Comment by HerbertJordan** [comment]
Hi Paul, just a few comments:
- please do not commit binary files (souffle-wave)
- please add a test case for the nullary relations (I couldn't find one)
- couldn't we use a RamFact instance with zero values as a replacement for RamNFact to avoid duplication?

**Comment by psubotic** [comment]
Hey, its a old version (I warned bernhard about this). I have to check in
the latest when 'I get to work in the morning. There is no RamNFact anymore
- its handled by a new NullValue class.

**Comment by psubotic** [comment]
also i have no idea why the wave gets checked in ... i will un add it i
guess ...

**Pull request closed without merging**